src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs

```diff
@@ -170,8 +127,23 @@ internal class EndpointConfig
         public string Name { get; set; }
         public string Url { get; set; }
         public HttpProtocols? Protocols { get; set; }
+
+        // REVIEW: ConfigSection doesn't seem comparable. If someone changes a custom key and consumes it in
```

Comment:
> It should be enumerable and a SequenceEquals style check should be adequate.

It would need to do recursive decent, right? We don't have any code, even in tests that reference the ConfigSection. I suspect no one does, so it feels like doing a lot for a little. Then again, I guess we should make this just work since we can.

> Is the old config section object still valid after a reload?

No. We use a new config section object.